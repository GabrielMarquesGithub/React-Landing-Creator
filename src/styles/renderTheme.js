import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from './theme';

export const renderTheme = (element) => {
  return render(<ThemeProvider theme={theme}>{element}</ThemeProvider>);
};
