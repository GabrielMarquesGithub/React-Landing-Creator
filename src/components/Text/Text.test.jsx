import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { Text } from '.';
import { theme } from '../../styles/theme';

describe('<Text />', () => {
  it('should render a text', () => {
    renderTheme(<Text>text</Text>);
    const text = screen.getByText('text');

    expect(text).toBeInTheDocument();
    expect(text).toHaveStyle({ 'font-size': theme.font.size.medium });
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<Text>text</Text>);
    expect(container).toMatchSnapshot();
  });
});
