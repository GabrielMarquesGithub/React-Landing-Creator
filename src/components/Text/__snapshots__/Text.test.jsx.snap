// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should match snapshot 1`] = `
<div>
  <p
    class="sc-bczRLJ fvsywh"
  >
    text
  </p>
</div>
`;
